### PR TITLE
[MONO][WASM] Changed wasmapp.LocalBuild.props to use net6.0

### PR DESCRIPTION
**Pull request by naricc**
The tfm in `WasmApp.LocalBuild.props` is used to build paths to the msbuild tasks, so it should match what `src/tasks/Directory.Build.props` has, which is `net6.0` currently.

Fixes:https://github.com/dotnet/runtime/issues/58906

### CODE REVIEW

**Comment by naricc** [comment]
@radical I have tested this with the dotnet/performance benchmarks, and it works. But I am not sure off all the local build scenarios that might also depend on this; are there any other things I should test locally?

**Comment by SamMonoRT** [comment]
@radical - please review this change. We want to make sure there are no regressions expected with this change to fix a few failing microbenchmarks.

**Comment by lewing** [comment]
> @radical - please review this change. We want to make sure there are no regressions expected with this change to fix a few failing microbenchmarks.

This will to be reverted as soon as the app host version is correctly updated.

**Comment by naricc** [comment]
@sampatel @lewing  Sorry, I linked the wrong issue (since updates). This is blocking all wasm/wasm-aot benchmark runs, not just a few.

File: src/mono/wasm/build/WasmApp.LocalBuild.props [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <Import Project="$(MSBuildThisFileDirectory)WasmApp.props" />
 
   <PropertyGroup>
-    <_NetCoreAppToolCurrent>net7.0</_NetCoreAppToolCurrent>
+    <_NetCoreAppToolCurrent>net6.0</_NetCoreAppToolCurrent>
```
Comment (radical) [inline src/mono/wasm/build/WasmApp.LocalBuild.props:25]: Rename this property to `_TargetFrameworkForNETCoreTasks` to be similar to what `src/tasks/Directory.Build.props` has, and so it's clear what this version should be.

Comment (lambdageek) [reply src/mono/wasm/build/WasmApp.LocalBuild.props:25]: @radical Why isn't this in `/Directory.Build.props` near other version props (or indeed defined in terms of one of those props)? https://github.com/dotnet/runtime/blob/main/Directory.Build.props#L43-L65

Comment (radical) [reply src/mono/wasm/build/WasmApp.LocalBuild.props:25]: The `*localBuild*` files get used outside the tree, like when building tests on helix.

**Review by radical — APPROVED** [review]

**Review by lewing — APPROVED** [review]

**Pull request merged**